### PR TITLE
[AIRFLOW-6223] BigQuery - improve hook tests [depends on AIRFLOW-6220]

### PR DESCRIPTION
**Pull request by TobKed**
This PR improve BQ hook tests by
- using BigQueryHook as a main hook
- unification mocking
- improving exception raising check
- improving assertions

Make sure you have checked _all_ steps below.

### Jira

- [ ] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-6223
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.
  - In case you are proposing a fundamental code change, you need to create an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)).
  - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).

### Description

- [ ] Here are some details about my PR, including screenshots of any UI changes:

### Tests

- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [ ] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### CODE REVIEW

File: tests/gcp/hooks/test_bigquery.py [hunk 1]
```diff
@@ -177,105 +119,146 @@ def test_invalid_syntax(self, table_input, var_name, exception_message):
...
                 "test_schema.json",
                 ["test_data.json"],
                 schema_update_options=['ALLOW_FIELD_ADDITION'],
                 write_disposition='WRITE_EMPTY'
             )
-        self.assertIn("schema_update_options is only", str(context.exception))
 
-    def test_cancel_queries(self):
+    @mock.patch("airflow.gcp.hooks.bigquery.BigQueryBaseCursor.poll_job_complete", side_effect=[False, True])
+    @mock.patch('airflow.gcp.hooks.base.CloudBaseHook._get_credentials_and_project_id')
```
Comment (mik-laj) [inline tests/gcp/hooks/test_bigquery.py]:
```suggestion
    @mock.patch('airflow.gcp.hooks.base.CloudBaseHook._get_credentials_and_project_id', return_value=("CREDENTIALS", "PROJECT_ID",))
```

File: tests/gcp/hooks/test_bigquery.py [hunk 2]
```diff
@@ -369,7 +386,9 @@ def test_run_load_with_non_csv_as_src_fmt(self, fmt):
 
 
 class TestTableDataOperations(unittest.TestCase):
-    def test_insert_all_succeed(self):
+    @mock.patch('airflow.gcp.hooks.base.CloudBaseHook._get_credentials_and_project_id')
```
Comment (mik-laj) [inline tests/gcp/hooks/test_bigquery.py]:
```suggestion
    @mock.patch('airflow.gcp.hooks.base.CloudBaseHook._get_credentials_and_project_id', return_value=("CREDENTIALS", "PROJECT_ID",))
```
For consistency

File: tests/gcp/hooks/test_bigquery.py [hunk 3]
```diff
@@ -397,15 +419,17 @@ def test_insert_all_succeed(self):
             body=body
         )
 
-    def test_insert_all_fail(self):
+    @mock.patch('airflow.gcp.hooks.base.CloudBaseHook._get_credentials_and_project_id')
```
Comment (mik-laj) [inline tests/gcp/hooks/test_bigquery.py]:
```suggestion
    @mock.patch('airflow.gcp.hooks.base.CloudBaseHook._get_credentials_and_project_id', return_value=("CREDENTIALS", "PROJECT_ID",))
```
For consistency

File: tests/gcp/hooks/test_bigquery.py [hunk 4]
```diff
@@ -416,14 +440,22 @@ def test_insert_all_fail(self):
...
+            r"BigQuery job failed\. Error was: 1 insert error\(s\) occurred: "
+            r"bq-project:bq_dataset\.bq_table\. Details: \[{'index': 1, 'errors': \[\]}\]"
+        ):
             cursor.insert_all(project_id, dataset_id, table_id,
                               rows, fail_on_error=True)
 
 
 class TestTableOperations(unittest.TestCase):
-    def test_create_view_fails_on_exception(self):
+    @mock.patch('airflow.gcp.hooks.base.CloudBaseHook._get_credentials_and_project_id')
```
Comment (mik-laj) [inline tests/gcp/hooks/test_bigquery.py]:
```suggestion
    @mock.patch('airflow.gcp.hooks.base.CloudBaseHook._get_credentials_and_project_id', return_value=("CREDENTIALS", "PROJECT_ID",))
```
For consistency

File: tests/gcp/hooks/test_bigquery.py [hunk 5]
```diff
@@ -432,16 +464,21 @@ def test_create_view_fails_on_exception(self):
...
+            resp=resp, content=b'Query is required for views')
+        mock_get_creds_and_proj_id.return_value = ("CREDENTIALS", project_id)
+        bq_hook = hook.BigQueryHook()
+        cursor = bq_hook.get_cursor()
+        with self.assertRaisesRegex(Exception, "HttpError 500 \"Query is required for views\""):
             cursor.create_empty_table(project_id, dataset_id, table_id,
                                       view=view)
 
-    def test_create_view(self):
+    @mock.patch('airflow.gcp.hooks.base.CloudBaseHook._get_credentials_and_project_id')
```
Comment (mik-laj) [inline tests/gcp/hooks/test_bigquery.py]:
```suggestion
    @mock.patch('airflow.gcp.hooks.base.CloudBaseHook._get_credentials_and_project_id', return_value=("CREDENTIALS", "PROJECT_ID",))
```
For consistency

File: tests/gcp/hooks/test_bigquery.py [hunk 6]
```diff
@@ -463,7 +502,9 @@ def test_create_view(self):
         }
         method.assert_called_once_with(projectId=project_id, datasetId=dataset_id, body=body)
 
-    def test_patch_table(self):
+    @mock.patch('airflow.gcp.hooks.base.CloudBaseHook._get_credentials_and_project_id')
```
Comment (mik-laj) [inline tests/gcp/hooks/test_bigquery.py]:
```suggestion
    @mock.patch('airflow.gcp.hooks.base.CloudBaseHook._get_credentials_and_project_id', return_value=("CREDENTIALS", "PROJECT_ID",))
```
For consistency

File: tests/gcp/hooks/test_bigquery.py [hunk 7]
```diff
@@ -514,7 +557,9 @@ def test_patch_table(self):
             body=body
         )
 
-    def test_patch_view(self):
+    @mock.patch('airflow.gcp.hooks.base.CloudBaseHook._get_credentials_and_project_id')
```
Comment (mik-laj) [inline tests/gcp/hooks/test_bigquery.py]:
```suggestion
    @mock.patch('airflow.gcp.hooks.base.CloudBaseHook._get_credentials_and_project_id', return_value=("CREDENTIALS", "PROJECT_ID",))
```
For consistency

File: tests/gcp/hooks/test_bigquery.py [hunk 8]
```diff
@@ -538,14 +585,18 @@ def test_patch_view(self):
             body=body
         )
 
-    def test_create_empty_table_succeed(self):
+    @mock.patch('airflow.gcp.hooks.base.CloudBaseHook._get_credentials_and_project_id')
```
Comment (mik-laj) [inline tests/gcp/hooks/test_bigquery.py]:
```suggestion
    @mock.patch('airflow.gcp.hooks.base.CloudBaseHook._get_credentials_and_project_id', return_value=("CREDENTIALS", "PROJECT_ID",))
```
For consistency

File: tests/gcp/hooks/test_bigquery.py [hunk 9]
```diff
@@ -562,7 +613,9 @@ def test_create_empty_table_succeed(self):
             body=body
         )
 
-    def test_create_empty_table_with_extras_succeed(self):
+    @mock.patch('airflow.gcp.hooks.base.CloudBaseHook._get_credentials_and_project_id')
```
Comment (mik-laj) [inline tests/gcp/hooks/test_bigquery.py]:
```suggestion
    @mock.patch('airflow.gcp.hooks.base.CloudBaseHook._get_credentials_and_project_id', return_value=("CREDENTIALS", "PROJECT_ID",))
```
For consistency

File: tests/gcp/hooks/test_bigquery.py [hunk 10]
```diff
@@ -605,20 +660,27 @@ def test_create_empty_table_with_extras_succeed(self):
             body=body
         )
 
-    def test_create_empty_table_on_exception(self):
+    @mock.patch('airflow.gcp.hooks.base.CloudBaseHook._get_credentials_and_project_id')
```
Comment (mik-laj) [inline tests/gcp/hooks/test_bigquery.py]:
```suggestion
    @mock.patch('airflow.gcp.hooks.base.CloudBaseHook._get_credentials_and_project_id', return_value=("CREDENTIALS", "PROJECT_ID",))
```
For consistency

File: tests/gcp/hooks/test_bigquery.py [hunk 11]
```diff
@@ -605,20 +660,27 @@ def test_create_empty_table_with_extras_succeed(self):
...
-        with self.assertRaises(Exception):
+            resp=resp, content=b'Bad request')
+        mock_get_creds_and_proj_id.return_value = ("CREDENTIALS", project_id)
+        bq_hook = hook.BigQueryHook()
+        cursor = bq_hook.get_cursor()
+        with self.assertRaisesRegex(Exception, "Bad request"):
             cursor.create_empty_table(project_id, dataset_id, table_id)
 
-    def test_get_tables_list(self):
+    @mock.patch('airflow.gcp.hooks.base.CloudBaseHook._get_credentials_and_project_id')
```
Comment (mik-laj) [inline tests/gcp/hooks/test_bigquery.py]:
```suggestion
    @mock.patch('airflow.gcp.hooks.base.CloudBaseHook._get_credentials_and_project_id', return_value=("CREDENTIALS", "PROJECT_ID",))
```
For consistency

File: tests/gcp/hooks/test_bigquery.py [hunk 12]
```diff
@@ -693,42 +762,75 @@ def run_with_config(config):
 
 
 class TestDatasetsOperations(unittest.TestCase):
+    @mock.patch(
+        'airflow.gcp.hooks.base.CloudBaseHook._get_credentials_and_project_id',
+        return_value=("CREDENTIALS", "PROJECT_ID",)
+    )
+    @mock.patch("airflow.gcp.hooks.bigquery.BigQueryHook.get_service")
+    def test_create_empty_dataset_no_dataset_id_err(self, mock_get_service, mock_get_creds_and_proj_id):
+        with self.assertRaisesRegex(ValueError, r"{} not provided datasetId\. Impossible to create dataset"):
```
Comment (mik-laj) [inline tests/gcp/hooks/test_bigquery.py]:
```
        with self.assertRaisesRegex(ValueError, r"?? not provided datasetId\. Impossible to create dataset"):
```
Is there good regexp in the marked place? It doesn't look good to me.

**Review by mik-laj — APPROVED** [review]

File: tests/gcp/hooks/test_bigquery.py (hunk 12 above)
Comment (TobKed) [reply tests/gcp/hooks/test_bigquery.py]: I agree it looks weird but the **Exception** message looked exactly like that. I've improved this message and adjusted regex as well.

**Comment by TobKed** [comment]
updated and rebased on the newest `master`

**Pull request merged**